Sources/Purchasing/Purchases/TransactionPoster.swift

```diff
@@ -92,15 +95,27 @@ final class TransactionPoster: TransactionPosterType {
 
         if systemInfo.dangerousSettings.internalSettings.usesStoreKit2JWS,
            let jwsRepresentation = transaction.jwsRepresentation {
-            self.fetchProductsAndPostReceipt(
-                transaction: transaction,
-                data: data,
-                receipt: .jws(jwsRepresentation),
-                completion: completion
-            )
+            if transaction.environment == .xcode, #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+                _ = Task<Void, Never> {
+                    let receipt = await transactionFetcher.receipt
+                    self.fetchProductsAndPostReceipt(
+                        transaction: transaction,
+                        data: data,
+                        receipt: .sk2receipt(receipt),
+                        completion: completion
+                    )
+                }
+            } else {
+                self.fetchProductsAndPostReceipt(
+                    transaction: transaction,
+                    data: data,
+                    receipt: .jws(jwsRepresentation),
+                    completion: completion
+                )
+            }
```

Comment:
This has so many branches is kinda hard to follow now. This new `if transaction.environent == .xcode` doesn't need to be inside of the `if let jwsRepresentation = transaction.jwsRepresentation`, right?

Maybe we can extract a method that produces the `EncodedAppleReceipt`, since all these branches end up doing the same:
```swift
self.fetchProductsAndPostReceipt(
    transaction: transaction,
    data: data,
    receipt: encodedReceipt,
    completion: completion
)
```